File: VolleyLib/src/main/java/com/kronos/volley/download/adapter/IModelObservable.java
package com.kronos.volley.download.adapter;

/**
 * Created by Leif Zhang on 16/8/18.
 * Email leifzhanggithub@gmail.com
 */
public interface IModelObservable {
    void registerDataSetObserver(IObserver observer);

    void unregisterDataSetObserver(IObserver observer);

    void notifyDataChange();
}

File: DownloadLib/src/main/java/com/kronos/download/adapter/IModelObservable.kt
package com.kronos.download.adapter

/**
 * Created by Leif Zhang on 16/8/18.
 * Email leifzhanggithub@gmail.com
 */
interface IModelObservable {
    fun registerDataSetObserver(observer: IObserver)

    fun unregisterDataSetObserver(observer: IObserver)

    fun notifyDataChange()

    fun unregisterAll()
}
